pkg/tenantctl/controller_test.go

```diff
@@ -124,7 +124,10 @@ func Test_Controller_syncHandler_AddsFinalizer(t *testing.T) {
 		// the tenant
 		fake.Tenant(tenantID, clientNSName),
 	)
-	ctl := NewController(cf, NewMetrics())
+
+	stopCh, ctl := startController(t, cf, false)
+	defer stopController(t, stopCh)
```

Comment:
I don't like that the controller gets started for those tests where we explicitly call e.g. `syncHandler()`.
Instead, just _create_ the controller as before and start `cf.CoreV1InformerFactory()` in addition.

You may use a helper function that returns a deferrable stop function:

```go
ctl := NewController(cf, NewMetrics())
defer startCoreV1Informer(ctl)()
```

```go
func startCoreV1Informer(ctl *Controller) func() {
	stopCh := make(chan struct{}, 0)
	ctl.factory.CoreV1InformerFactory().Start(stopCh)
	return func() {
		stopCh <- struct{}{}
	}
}
```